pyproject.toml

```diff
@@ -41,7 +41,7 @@ dependencies    = [
     # Also upgrade requirements_test_min.txt.
     # Pinned to dev of second minor update to allow editable installs and fix primer issues,
     # see https://github.com/pylint-dev/astroid/issues/1341
-    "astroid>=3.0.0a8,<=3.1.0-dev0",
+    "astroid @ git+https://github.com/pylint-dev/astroid.git@cf8763a2b8e897ec7c8389906f3cb13714300cd2",
```

Comment:
We can then immediately go up to 3.0.0a9 next, but I wanted to do baby steps first in case more changes are needed.